edgelet/edgelet-image-cleanup/src/image_gc.rs

```diff
@@ -0,0 +1,263 @@
+// This image is used as a fallback in certain scenarios and we have to make sure that we never delete it
+// (though the customer can still do so).
+//
+// To wit, there are 4 possibilities (pertaining to this method's return values):
+//    Image Id found       Has image been deleted?     Interpretation
+//        yes                        yes               N/A [can't happen]
+//        yes                        no                prune images, as per usual
+//        no                         yes               customer deleted bootstrap; prune images, as per usual
+//        no                         no                (Implicit) Docker Engine API error; update persistence file but
+//                                                     do not prune images to ensure EA bootstrap isn't deleted
+
+async fn get_bootstrap_image_id(
+    runtime: &edgelet_docker::DockerModuleRuntime<http_common::Connector>,
+    edge_agent_bootstrap: String,
+) -> Result<(String, bool), ImageCleanupError> {
+    let is_bootstrap_deleted: bool = false;
+
+    let image_name_to_id = ModuleRuntime::list_images(runtime)
+        .await
+        .map_err(ImageCleanupError::ListImages)?;
+    let image_id_option = image_name_to_id.get(&edge_agent_bootstrap);
+
+    let bootstrap_image_id =
+        image_id_option.map_or_else(String::default, |image_id| image_id.to_string());
+
+    Ok((bootstrap_image_id, is_bootstrap_deleted))
+}
+
+fn validate_settings(settings: &ImagePruneSettings) -> Result<(), ImageCleanupError> {
+    if settings.cleanup_recurrence() < MIN_CLEANUP_RECURRENCE {
+        return Err(ImageCleanupError::InvalidConfiguration(
+            "cleanup recurrence cannot be less than 1 day".to_string(),
+        ));
+    }
+
+    let times = NaiveTime::parse_from_str(&settings.cleanup_time(), "%H:%M");
+    if times.is_err() {
+        return Err(ImageCleanupError::InvalidConfiguration(
+            "Invalid cleanup time. Expected format is \"HH:MM\" in 24-hour format.".to_string(),
+        ));
+    }
+
+    Ok(())
+}
+
+fn get_sleep_time_mins(times: &str) -> u32 {
+    // if string is empty, or if there's an input error, we fall back to default (midnight)
+    let cleanup_mins = if times.is_empty() {
+        TOTAL_MINS_IN_DAY
+    } else {
+        let cleanup_time: Vec<&str> = times.split(':').collect();
+        let hour = cleanup_time.get(0).unwrap().parse::<u32>().unwrap();
+        let minute = cleanup_time.get(1).unwrap().parse::<u32>().unwrap();
+
+        60 * hour + minute
+    };
+
+    let current_hour = chrono::Local::now().hour();
+    let current_minute = chrono::Local::now().minute();
+    let current_time_in_mins = 60 * current_hour + current_minute;
+
+    if current_time_in_mins < cleanup_mins {
+        cleanup_mins - current_time_in_mins
+    } else {
+        TOTAL_MINS_IN_DAY - (current_time_in_mins - cleanup_mins)
+    }
+}
+
```

Comment:
```suggestion
```

Reply:
Is the suggested change to remove this method entirely?